File: dropwizard/src/main/java/com/yammer/dropwizard/jersey/params/BooleanParam.java
package com.yammer.dropwizard.jersey.params;

// TODO: 11/14/11 <coda> -- document BooleanParam
// TODO: 11/14/11 <coda> -- test BooleanParam

public class BooleanParam extends AbstractParam<Boolean> {
    public BooleanParam(String input) {
        super(input);
    }

    @Override
    protected String errorMessage(String input, Exception e) {
        return '"' + input + "\" must be \"true\" or \"false\"";
    }

    @Override
    protected Boolean parse(String input) {
        return Boolean.valueOf(input);
    }
}

File: dropwizard-scala_2.9.1/src/main/scala/com/yammer/dropwizard/scala/params/BooleanParam.scala
package com.yammer.dropwizard.scala.params

object BooleanParam {
  def apply(value: Boolean): BooleanParam = BooleanParam(value.toString)
}

/**
 * Parses "true" and "false" to Boolean values.
 */
case class BooleanParam(s: String) extends AbstractParam[Boolean](s) {
  protected def parse(input: String) = input.toBoolean

  override protected def renderError(input: String, e: Throwable) =
    "Invalid parameter: %s (Must be \"true\" or \"false\".)".format(input)
}
